### PR TITLE
Add FAQ when "401 Unauthorized" from registry

### PR DESCRIPTION
**Pull request by chanseokoh**
For #1619.

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -343,6 +344,22 @@ See more at [Using Google Container Registry (GCR) with Minikube](https://ryanes
 
 Jib currently requires configuring your build tool to use the appropriate [Java networking properties](https://docs.oracle.com/javase/8/docs/api/java/net/doc-files/net-properties.html) (`https.proxyHost`, `https.proxyPort`, `https.proxyUser`, `https.proxyPassword`).
 
+### What should I do when the registry responds with UNAUTHORIZED?
+
+When the registry returns "401 Unauthorized" and/or `"code":"UNAUTHORIZED"`, most of the time it genuinely means the registry does not see (yet) that you have the right to pull/push images. This is often due to credential misconfiguration. Examples:
+
+* You did not configure auth information in the places where Jib searches for by default.
```
Comment (loosebazooka) [inline docs/faq.md]:
```diff
+ You did not configure auth information in the places where Jib searches for by default.
- You did not configure auth information in the default places Jib searches.
```

File: docs/faq.md [hunk 2]
```diff
@@ -343,6 +344,22 @@ See more at [Using Google Container Registry (GCR) with Minikube](https://ryanes
...
+   - `$HOME/.docker/config.json`, [one of the configuration files](https://docs.docker.com/engine/reference/commandline/cli/#configuration-files) for the `docker` command line tool. Check the linked document, the [credential store](https://docs.docker.com/engine/reference/commandline/login/#credentials-store) and [credential helper](https://docs.docker.com/engine/reference/commandline/login/#credential-helpers) sections, and [this](https://github.com/GoogleContainerTools/jib/issues/101) for how to configure auth. For example, you can do `docker login` to save auth in `config.json`, but it is often recommended to configure a credential helper (also configurable in `config.json`).
```
Comment (loosebazooka) [inline docs/faq.md]: I'm not sure what "Check the linked document" is referencing. But you can just replace it with "See: "

File: docs/faq.md [hunk 3]
```diff
@@ -343,6 +344,22 @@ See more at [Using Google Container Registry (GCR) with Minikube](https://ryanes
...
+      - Specific credentials (not recommend): [`<from/to><auth><username>/<password>`](https://github.com/GoogleContainerTools/jib/tree/master/jib-maven-plugin#using-specific-credentials) or in [`settings.xml`](https://github.com/GoogleContainerTools/jib/tree/master/jib-maven-plugin#using-maven-settings) for Maven / [`from/to.auth.username/password`](https://github.com/GoogleContainerTools/jib/tree/master/jib-gradle-plugin#using-specific-credentials) for Gradle
+      - These parameters can also be set through properties: [Maven](https://github.com/GoogleContainerTools/jib/tree/master/jib-maven-plugin#system-properties) / [Gradle](https://github.com/GoogleContainerTools/jib/tree/master/jib-gradle-plugin#system-properties)
+* You configured a credential helper, but the helper is not on `$PATH`. This is especially common when running Jib inside IDE where the IDE binary is launched directly from an OS menu.
```
Comment (loosebazooka) [inline docs/faq.md]: .... OS menu +and does not have access to your shell's environment.

File: docs/faq.md [hunk 4]
```diff
@@ -343,6 +344,22 @@ See more at [Using Google Container Registry (GCR) with Minikube](https://ryanes
...
+* Typo in username or password. Sometimes wrong registry names for the given auth information.
```
Comment (loosebazooka) [inline docs/faq.md]: "Typo" -> "A typo" or "Typos"

I'm not sure what the sentence "Sometimes wrong registry names for the given auth information." means?

File: docs/faq.md [hunk 5]
```diff
@@ -343,6 +344,22 @@ See more at [Using Google Container Registry (GCR) with Minikube](https://ryanes
...
+
+If you encounter other issues interacting with a registry than `UNAUTHORIZED`, check ["How can I diagnose problems pulling or pushing from remote registries?"](#how-can-i-diagnose-problems-pulling-or-pushing-from-remote-registries).
```
Comment (loosebazooka) [inline docs/faq.md]:
```diff
- If you encounter other issues interacting with a registry than `UNAUTHORIZED`
+ If you encounter issues interacting with a registry other than `UNAUTHORIZED`
```

File: docs/faq.md (hunk 4 above)
Comment (chanseokoh) [reply docs/faq.md]: +1. Will go with "Typos in username, password, image names, or registry names."

**Comment by chanseokoh** [comment]
Awesome. They sound better.

**Comment by chanseokoh** [comment]
Just came up with more ideas:

- Insecure private registry. Jib does not send auth over HTTP.
- Different auth configurations in multiple places, and Jib is not picking up the desired one due to config priority

**Pull request merged**